### PR TITLE
Award 12 in 23 badge

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/models/badges/participant_in_12_in_23_badge.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    def award_to?(user)
+      challenge = user.challenges.find_by(challenge_id: User::Challenge::CHALLENGE_12_IN_23)
+      return false if challenge.nil?
+
+      user.solutions.completed.joins(exercise: :track).
+        where('completed_at >= ?', challenge.created_at).
+        group(:track_id).
+        having("count(*) >= 5").
+        exists?
```
Comment (ErikSchierboom) [inline app/models/badges/participant_in_12_in_23_badge.rb]: This is quite slow (1s), due to there not being an appropriate index.

Comment (iHiD) [reply app/models/badges/participant_in_12_in_23_badge.rb]: Hmm, it's only 5ms when I check...

Screenshot 2023-01-31 at 22 55 23

Comment (ErikSchierboom) [reply app/models/badges/participant_in_12_in_23_badge.rb]: Probably depends on the number of solutions I guess

Comment (iHiD) [reply app/models/badges/participant_in_12_in_23_badge.rb]: I guess. That's your user above though, who has a lot!

Comment (ErikSchierboom) [reply app/models/badges/participant_in_12_in_23_badge.rb]: True. Ah well

**Review by iHiD — APPROVED** [review]

**Pull request merged**